lib/ansible/plugins/lookup/dict.py

```diff
@@ -70,7 +70,7 @@ def run(self, terms, variables=None, **kwargs):
         for term in terms:
             # Expect any type of Mapping, notably hostvars
             if not isinstance(term, Mapping):
-                raise AnsibleError("with_dict expects a dict")
+                raise AnsibleError(f"with_dict expects a dict, got '{term}' of type {type(term)})")
```

Comment:
As long as we're improving things: the lookup is just called `dict`. `with_dict` is one way to use the `dict` lookup, but not the only one.